MAUI/MauiAppWithBroker/MSALClient/PCAWrapper.cs

```diff
@@ -30,16 +38,30 @@ public class PCAWrapper
         // private constructor for singleton
         private PCAWrapper()
         {
+            var assembly = Assembly.GetExecutingAssembly();
+            using var stream = assembly.GetManifestResourceStream("MauiAppWithBroker.appsettings.json");
+            AppConfiguration  = new ConfigurationBuilder()
+                .AddJsonStream(stream)
+                .Build();
+
             // Create PublicClientApplication once. Make sure that all the config parameters below are passed
             PCA = PublicClientApplicationBuilder
-                                        .Create(AppConstants.ClientId)
-                                        .WithTenantId(AppConstants.TenantId)
+                                        .Create(AppConfiguration["AzureAd:ClientId"])
+                                        .WithTenantId(AppConfiguration["AzureAd:TenantId"])
                                         .WithExperimentalFeatures() // this is for upcoming logger
                                         .WithLogging(_logger)
                                         .WithBroker()
                                         .WithRedirectUri(PlatformConfig.Instance.RedirectUri)
                                         .WithIosKeychainSecurityGroup("com.microsoft.adalcache")
                                         .Build();
+
+            if (DeviceInfo.Current.Platform == DevicePlatform.WinUI)
+            {
+                //Cache configuration and hook-up to public application. Refer to https://github.com/AzureAD/microsoft-authentication-extensions-for-dotnet/wiki/Cross-platform-Token-Cache#configuring-the-token-cache
+                var storageProperties = new StorageCreationPropertiesBuilder(AppConfiguration["CacheFileName"], AppConfiguration["CacheDir"]).Build();
+                MsalCacheHelper.CreateAsync(storageProperties)
+                    .ContinueWith(async msalCacheHelper => (await msalCacheHelper).RegisterCache(PCA.UserTokenCache));

```

Comment:
Not happy to see sync over async. Singleton pattern is not good here (and in general singleton has problems). @SameerK-MSFT - can you suggest a solution?

I'd recommend using a normal class with async initialization.

Reply:
I was thinking the same thing...

What I could also do here is disable the application until the loading is complete. I.e. disable sign-ins and run this process in the background.